### PR TITLE
[Docs]Add overridepolicy guide for users

### LINKED ISSUE #1251: [docs] OverridePolicy documents
Add OverridePolicy user guide based on https://github.com/karmada-io/karmada/blob/master/docs/override-policy.md.

**What would you like to be added**:

**Why is this needed**:

/assign

### PR DESCRIPTION
**Pull request by Poor12**
Signed-off-by: Poor12 <shentiecheng@huawei.com>

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Add overridepolicy guide for users. Deprecate OverrideSpec.Overriders and use OverrideSpec.OverrideRules instead.

**Which issue(s) this PR fixes**:
Fixes #1251 

**Special notes for your reviewer**:
None

### CODE REVIEW

File: docs/override-policy.md [hunk 1]
```diff
@@ -4,11 +4,152 @@ The [OverridePolicy][1] and [ClusterOverridePolicy][2] are used to declare overr
 they are propagating to different clusters.
 
 ## Difference between OverridePolicy and ClusterOverridePolicy
-TBD
+ClusterOverridePolicy represents the cluster-wide policy that overrides a group of resources to one or more clusters while OverridePolicy will apply to resources in the same namespace as the namespace-wide policy. For cluster scoped resources, apply ClusterOverridePolicy by policies name in ascending. For namespaced scoped resources, first apply ClusterOverridePolicy, then apply OverridePolicy.
+
 ## Resource Selector
-TBD
+
+ResourceSelectors restricts resource types that this override policy applies to. If you ignore this field it means matching all clusters.
```
Comment (RainbowMango) [inline docs/override-policy.md:11]: typo `all clusters`?

File: docs/override-policy.md [hunk 2]
```diff
@@ -4,11 +4,152 @@ The [OverridePolicy][1] and [ClusterOverridePolicy][2] are used to declare overr
...
+      kind: Deployment
+      name: nginx
+      namespace: test
+      labelSelector:
+        matchLabels:
+          app: nginx
+  overrideRules:
+  ...
+```
+It means override rules below will only be applied to `Deployment` which is named nginx in test namespace and has labels with `app: nginx`.
```
Comment (RainbowMango) [inline docs/override-policy.md:37]:
```suggestion
It means override rules above will only be applied to `Deployment` which is named `nginx` in `test` namespace and has labels with `app: nginx`.
```

File: docs/override-policy.md [hunk 3]
```diff
@@ -4,11 +4,152 @@ The [OverridePolicy][1] and [ClusterOverridePolicy][2] are used to declare overr
...
+  ...
+  overrideRules:
+    - targetCluster:
+        labelSelector:
+          matchLabels:
+            cluster: member1 
+      overriders:
+      ...
+```
+It means override rules below will only be applied to those resources propagated to clusters which has `cluster: member1` label.
```
Comment (RainbowMango) [inline docs/override-policy.md:67]:
```suggestion
It means override rules above will only be applied to those resources propagated to clusters which has `cluster: member1` label.
```

**Review by RainbowMango — COMMENTED** [review]
cc @Tingtal Could you please help with it?

**Comment by Poor12** [comment]
Fix some typo problems.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/apporve

Thanks @Poor12 , I'll make some changes based on your PR. Such as the `.spec.TargetCluster` and `.spec.Overriders` has been deprecated.

**Pull request merged**